pkg/util/constants.go

```diff
@@ -28,6 +28,9 @@ const (
 
 	// ManagedByKarmadaLabelValue indicates that resources are managed by karmada controllers.
 	ManagedByKarmadaLabelValue = "true"
+
+	// ResourceScaledByHPALabel is added to mark resource template scaled by HPA, then its replicas will be retained when propagating.
+	ResourceScaledByHPALabel = "horizontalpodautoscaler.karmada.io/name"
```

Comment:
Can we set the key more specific so that users can understand the meaning of the key as soon as they see it?

Reply:
so, how? I may not got your mean

Reply:
how about: 

```go
// ResourceScaledByHPALabel is added to mark resource template scaled by HPA, possible conflict-resolution of replicas: retain-membercluster
ResourceScaledByHPALabel = "resolve-replicas-conflict-when-scaled-by-hpa"
```

i mean `resolve-replicas-conflict-when-scaled-by-hpa: retain-membercluster`

Reply:
Is it not limited to hpa? If the user adds this label, the function will also take effect. Therefore, we can focus on the function.

Reply:
How about `resolve-replicas-conflict: retain`

Reply:
cc @RainbowMango

Reply:
I think a label should contain the Karmada domain name so that people can figure out if it is introduced/added by Karmada.

Propose another one:
```goland
	// RetainReplicas is a reserved label to indicate whether the replicas should be retained.
	RetainReplicas = "resourcetemplate.karmada.io/retain-replicas"
```

Given this label is oriented at `resource template`, so it could be organized to `resourcetemplate.karmada.io`.

E.g.

```
resourcetemplate.karmada.io/retain-replicas: true   // with value `true` indicates retain
resourcetemplate.karmada.io/retain-replicas: false  // with value `false` and others, indicates not retain
```

@chaunceyjiang What do you think?